---
title: "Comparing Influenza A H1N1 and H3N2 Genetic Drift 2013-2017, US"
subtitle: "Molecular Epidemiolgy Final Report, Spring 2024"
author: "Savannah Hammerton"
format: 
   docx:
    citeproc: false
    filters:
      - at: pre-render
        path: "../../_extensions/andrewheiss/wordcount/citeproc.lua"
      - at: pre-render
        path: "../../_extensions/andrewheiss/wordcount/wordcount.lua"
csl: emerging-infectious-diseases.csl
bibliography: project.bib
abstract: |
  Genetic drift leads a need to update influenza vaccine strains most seasons. Distance measures over time can tell us how much drift has occurred. We aimed to determine if H1N1 or H3N2 appeared to drift more over time, and found that there was no significant difference between the subtypes. 
---

\newpage

# Introduction 

Genetic drift leads a regular need to re-assess and update influenza vaccine strains from year to year [@cdc22a]. Differences between influenza vaccine strains and circulating strains have been cited as one of the causes of reduced vaccine efficacy [@belongia16; @dacosta22; @tenforde21; @tricco13]. "Mismatch" between the strains is often defined somewhat arbitrarily and dichotomously, but using continuous genetic distance measures could lead to a better understanding of the impact of genetic drift and vaccine mismatch on changes in vaccine effectiveness in populations [@tricco13; @dacosta22]. However, there is not just one circulating strain per season, as genetic drift happens continuously over time and around the globe. One way we can examine and compare genetic drift is to assess the gradual change and evolution over time and compare genetic distances using phylogenetic tree-based methods. In this analysis, we used such methods to compare genetic drift in H1N1 and H3N2 Influenza A subtypes in the United States from 2013 to 2017. 

# Methods 

Data were obtained from [BV-VRC.org](https://www.bv-brc.org/view/Taxonomy/11320?and(eq(host_group,Human),eq(isolation_country,%22USA%22),or(eq(collection_year,%222016%22),eq(collection_year,%222015%22),eq(collection_year,%222014%22),eq(collection_year,%222013%22),eq(collection_year,%222017%22)),or(eq(subtype,%22H1N1%22),eq(subtype,%22H3N2%22)),eq(genome_status,%22Complete%22),eq(segment,%224%22))#view_tab=genomes&filter=false) using the filters applied in the attached link [@bvbrc]. The data were loaded and cleaned in R (FASTA file was loaded using the `seqinr` package) [@charif07; @rcoreteam24], and fifty random samples were taken in each subtype of interest in each year to reduce the final sample size from 9,853 to 500 (seed was set to 42 for reproducibility). The `msa` package was then used perform multiple sequence alignment and prepare data for modeling [@bodenhofer15]. 

Unweighted Pair Group Method with Arithmetic Mean (UPGMA) [@moulton18] was used to construct a rooted tree and bootstrapped estimates were calculated for edge lengths using the `phangorn` R package [@schliep11]. These estimates were then used to calculate mean total distances and 95% confidence intervals (CIs) for H1N1 and H3N2 over the study period, as well as a contrast between the two ($D_{H1N1} - D_{H3N2}$). This contrast was used to test the hypotheses that H3N2 phylogenetic trees would exhibit increased diversity and genetic distance compared to H1N1 trees (based on results from exploratory analysis). 

Maximum likelihood methods were then used to select and fit models and generate final phylogenetic tree models using the `phangorn` R package [@schliep11]. The JC, K80, and SYM models were tested with all models also testing base, +I, +G, and G+I versions. Final models were selected based on a combination of logLikelihood, AIC, and BIC measures. Collection date was used to root the models in both subtypes. Specific collection dates were missing in five H1N1 and two H3N2 records (only years provided). In these cases, the first of the reported year was used as the date. 


# Results

Full bootstrap distributions of distance measures for H1N1 and H3N2 over the study period can be seen in @fig-dists. Distance and contrast bootstrapped estimates and 95% CIs are shown in @tbl-contrast. Visually, it appears as though H3N2 had a slight trend towards higher distance in the full distributions. However, when examining the final estimates, and in particular, the contrast estimates, we see that the 95% CI crosses zero, and we must thus conclude that there is no statistically significant difference in the total amount of genetic drift over time in our samples over the study period. 

```{r dists}
#| echo: FALSE 
#| message: false
#| warning: false
#| label: fig-dists
#| fig-cap: "Bootstrap distributions of total distances for H1N1 (red) and H3N2 (blue) subtypes over the study period."
knitr::include_graphics(here::here("results/figures/dist_fig.png"))
```

```{r contrast}
#| echo: FALSE 
#| message: false
#| warning: false
#| label: tbl-contrast
#| tbl-cap: "Bootstrapped means and 95% confidence intervals (CIs) for total H1N1 and H3N2 tree distances and contrasts. As the 95% CI for the contrast crosses zero, we conclude there is no statistically significant difference in the distances of the two sub-types, and thus no significant difference in the antigenic drift over the analysis period."

readRDS(here::here("results/tables/res.rds")) |> 
  gt::gt() |> 
  gt::fmt_number(decimals = 2) |> 
  gt::as_raw_html()
```

Models were tested as described above, and fit statistics are presented in @tbl-mods. The SYM+I+G models for both H1N1 and H3N2 subtypes appeared to fit generally better when balancing logLikelihood, AIC, and BIC values, so this model was chosen for both subtypes. The same model was intentionally chosen for each subtype to make the resulting trees more comparable. The final trees are presented in @fig-trees. While initial results revealed that overall distance was similar across the two subtypes, H1N1 appeared to have a more consistent, general evolution, while H3N2 evolution appeared much more sporadic and random. 


```{r}
#| echo: FALSE 
#| message: false
#| warning: false
#| label: tbl-mods
#| tbl-cap: "Fit statistics for H1N1 (top) and H3N2 (bottom) tested phlylogenetic tree models."
readRDS(here::here("results/tables/mods_tab.rds")) |> 
  gt::fmt_number(decimals = 1) |> 
  gt::as_raw_html()
```


```{r}
#| echo: FALSE 
#| message: false
#| warning: false
#| label: fig-trees
#| fig-cap: "SYM+G+I trees rooted by collection date for H1N1 (left) and H3N2 (right)."
knitr::include_graphics(here::here("results/figures/trees.png"))
```

# Conclusions

This analysis aimed to compare genetic drift and evolution of influenza A H1N1 and H3N2 in the United States over the 2013-2017 time period. Both distance-based and maximum likelihood models were used to compare the changes in randomly sampled sequences collected over that period. While we expected to see increased change in H3N2 over H1N1 based on exploratory results, we found no statistically significant difference in the total H1N1 and H3N2 distances. However, final selected models showed a more consistent evolution in H1N1 strains compared to H3N2 strains, which appeared more random. This matches literature that reports H3N2 generally evolves quickly [@potter19]. Additionally, it is possible that the study period taking place a few years after the 2009 H1N1 pandemic also impacted the more gradual and consistent evolution of those strains [@wilasang22]. 

This analysis was limited in computational resources, requiring smaller sample sizes and less complex models. Future analyses could benefit from further examination using larger samples over longer study periods to assess if one subtype appears to drift more rapidly than the other. Sub-analyses could also stratify by different, shorter time periods to examine how the evolution rates change over time and interact with each other. 

\newpage

# References 
